webui/react/src/components/ExperimentActions.tsx

```diff
@@ -0,0 +1,169 @@
+import { Button, Popconfirm } from 'antd';
+import React, { useCallback, useState } from 'react';
+
+import { archiveExperiment, killExperiment, launchTensorboard, setExperimentState,
+} from 'services/api';
+import { ExperimentDetails, RunState, TBSourceType } from 'types';
+import { openCommand } from 'utils/routes';
+import { cancellableRunStates, killableRunStates, terminalRunStates } from 'utils/types';
+
+import css from './ExperimentActions.module.scss';
+
+interface Props {
+  experiment: ExperimentDetails;
+  finally: () => void; // A callback to trigger after an action is done.
+}
+
+const ExperimentActions: React.FC<Props> = ({ experiment, finally: updateFn }: Props) => {
+
+  interface ButtonLoadingStates {
+    kill: boolean;
+    [RunState.StoppingCanceled]: boolean;
+    archive: boolean;
+    [RunState.Paused]: boolean;
+    [RunState.Active]: boolean;
+    tsb: boolean; // tensorboard
+  }
+
+  const [ buttonStates, setButtonStates ] = useState<ButtonLoadingStates>({
+    [RunState.StoppingCanceled]: false,
+    archive: false,
+    [RunState.Active]: false,
+    kill: false,
+    [RunState.Paused]: false,
+    tsb: false,
+  });
+
+  const killExperimentCB = useCallback(() => {
+    setButtonStates(state => ({ ...state, kill: true }));
+    killExperiment({ experimentId: experiment.id })
+      .then(updateFn)
+      .finally(() => setButtonStates(state => ({ ...state, kill: false })));
+  }, [ experiment.id, updateFn ]);
+
+  const archiveCB = useCallback((archive: boolean) =>
```

Comment:
**nit:** `handleArchive` and maybe move it above `handleKill` for alphabetical ordering?